edb/schema/objects.py

```diff
@@ -1101,7 +1104,15 @@ def _prepare_id(
         try:
             return get_known_type_id(name)
         except errors.SchemaError:
-            return cls.generate_id(schema, data)
+            if stdmode:
+                # When compiling the standard library, we generate
+                # stable ids based on the internal name and the type's
+                # name. This keeps std schemas compatible across
+                # minor versions at least.
+                return uuidgen.uuid5(
+                    TYPE_ID_NAMESPACE, f'{name}-{cls.__name__}')
+            else:
+                return cls.generate_id(schema, data)
```

Comment:
We don't seem to actually overload `generate_id` anywhere anymore, do you want to get rid of it while you're here?

Reply:
Or, if you want to keep it for some reason, I'd push the `stdmode` thing into it.

Reply:
Yeah we'll just drop it